### ISSUE TITLE
Remote database is deleted on first "Apply and send" 

### ISSUE THREAD

**Issue by trmckay** [vrtmrz/obsidian-livesync#61]
I've been using this plugin for a month or so now and recently ran into some issues. I'm not sure if this was triggered by an update but I noticed my notes had stopped syncing.

I wasn't able to fix it so I backed up my markdown and started over. However, I am unable to setup the mobile app with my server.

# Summary

When configuring mobile app then selecting "Apply and send", the data is synced to the server, then the database is deleted. This results in subsequent API calls returning 404.

# Details

- Obsidian version: 1.2.0 (50), API v0.14.2
- Plugin version: 0.8.2

`docker-compose.yml` for couchdb:

```
version: "3.7"

services:
    couchdb:
        image: couchdb
        container_name: obsidian-couchdb
        environment:
            - COUCHDB_USER=****
            - COUCHDB_PASSWORD=****
        volumes:
            - ./data/couchdb:/opt/couchdb/data
            - ./conf/local.ini:/opt/couchdb/etc/local.ini
        networks:
            - proxy
        restart: unless-stopped

networks:
  proxy:
    external: true
```

This is fronted by a Caddy reverse-proxy configured like so:

```
obsidian-db.mydomain.com {
  handle_path /* {
    respond "" 403
  }
  handle_path /e=****/* {
    reverse_proxy obsidian-couchdb:5984
  }
}
```

# Steps to reproduce

1. Create the database.

```
$ https --session=obsidian-db PUT obsidian-db.mydomain.com/e=****/notes
HTTP/1.1 201 Created
Cache-Control: must-revalidate
Content-Length: 12
Content-Type: application/json
Date: Wed, 20 Apr 2022 00:36:57 GMT
Location: https://obsidian-db.mydomain.com/notes
Server: Caddy, CouchDB/3.2.1 (Erlang OTP/23)
X-Couch-Request-Id: 4f6d11f144
X-Couchdb-Body-Time: 0

{
    "ok": true
}
```

```
$ https --session=obsidian-db GET obsidian-db.mydomain.com/e=***/notes        
HTTP/1.1 200 OK
Cache-Control: must-revalidate
Content-Length: 453
Content-Type: application/json
Date: Wed, 20 Apr 2022 00:38:13 GMT
Server: Caddy, CouchDB/3.2.1 (Erlang OTP/23)
X-Couch-Request-Id: 9224fc3b6a
X-Couchdb-Body-Time: 0

{
    "cluster": {
        "n": 1,
        "q": 2,
        "r": 1,
        "w": 1
    },
    "compact_running": false,
    "db_name": "notes",
    "disk_format_version": 8,
    "doc_count": 0,
    "doc_del_count": 0,
    "instance_start_time": "0",
    "props": {},
    "purge_seq": "0-g1AAAABPeJzLYWBgYMpgTmHgzcvPy09JdcjLz8gvLskBCeexAEmGBiD1HwiyEhlwqEtkSKqHKMgCAIT2GV4",
    "sizes": {
        "active": 0,
        "external": 0,
        "file": 16692
    },
    "update_seq": "0-g1AAAACLeJzLYWBgYMpgTmHgzcvPy09JdcjLz8gvLskBCeexAEmGBiD1HwiyMpgTGXKBAuxpaYbmSeap6HpwmJLIkFSPoj3FJNXEMjkNXXEWACNhKx8"
}
```

2. Test connection on mobile app; response: `Connected to notes`. At the same time, one can see couchdb returning a 200 to for the `GET /notes` request from the app.

3. "Apply and send" from the mobile app.

Here are the corresponding couchdb logs after applying the configuration:

```
rbsidian-couchdb | [notice] 2022-04-20T00:42:03.678580Z nonode@nohost <0.15267.0> 457803bc30 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 200 ok 4
obsidian-couchdb | [notice] 2022-04-20T00:42:03.703950Z nonode@nohost <0.15267.0> 4447edc1f5 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 200 ok 4
obsidian-couchdb | [notice] 2022-04-20T00:42:03.733413Z nonode@nohost <0.15267.0> c8557bfec7 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/obsydian_livesync_version? 404 ok 2
obsidian-couchdb | [notice] 2022-04-20T00:42:03.755440Z nonode@nohost <0.15267.0> 0916f5c7db obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/obsydian_livesync_version? 404 ok 2
obsidian-couchdb | [notice] 2022-04-20T00:42:03.807252Z nonode@nohost <0.15267.0> 67f1c35203 obsidian-db.mydomain.com 129.65.145.196 obsidian PUT /notes/obsydian_livesync_version 201 ok 32
obsidian-couchdb | [notice] 2022-04-20T00:42:03.827524Z nonode@nohost <0.15267.0> d41212f0e3 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/_local/obsydian_livesync_milestone? 404 ok 2
obsidian-couchdb | [notice] 2022-04-20T00:42:03.907242Z nonode@nohost <0.15267.0> 7560ef061d obsidian-db.mydomain.com 129.65.145.196 obsidian PUT /notes/_local/obsydian_livesync_milestone 201 ok 58
obsidian-couchdb | [notice] 2022-04-20T00:42:03.940475Z nonode@nohost <0.15267.0> d09e06acdf obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 200 ok 4
obsidian-couchdb | [notice] 2022-04-20T00:42:03.965198Z nonode@nohost <0.15267.0> 9f3b731515 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 200 ok 4
obsidian-couchdb | [notice] 2022-04-20T00:42:03.999035Z nonode@nohost <0.15267.0> 1fb2619c64 obsidian-db.mydomain.com 129.65.145.196 obsidian DELETE /notes/ 200 ok 15
obsidian-couchdb | [notice] 2022-04-20T00:42:04.044089Z nonode@nohost <0.15267.0> b8de6814a2 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 404 ok 2
obsidian-couchdb | [notice] 2022-04-20T00:42:04.080332Z nonode@nohost <0.15267.0> ec7de63508 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 404 ok 3
obsidian-couchdb | [notice] 2022-04-20T00:42:04.115652Z nonode@nohost <0.15267.0> 3529d20111 obsidian-db.mydomain.com 129.65.145.196 obsidian GET /notes/ 404 ok 3
```

And a view from the app:

![IMG_AD4C63F1AC93-1](https://user-images.githubusercontent.com/43476566/164124979-55e20564-414a-4ad2-acd8-9b718a5d8a61.jpeg)

**Comment by trmckay** [comment]
Of course I find this minutes after submitting the issue despite trying all sorts of things beforehand. Enabling "Connect with the old method" solves this issue. Is this the intended behavior?

**Comment by vrtmrz** [comment]
Thank you for very detailed information! It so helped me.

Deleting the remote database (once) is normal behavior on changing the passphrase. (Otherwise, we will leave the histories of entries that were saved with plain text or encrypted with a compromised passphrase)
And, it should be created again automatically on the first replication by detecting 404.

But, the new API wasn't able to detect 404 completely (#54 was not the straight way.). Thank you for making the issue.

I fixed this in v0.8.3!

**Comment by trmckay** [comment]
Awesome! Thank you for the fast update and a super useful plugin! Just tested with 0.8.3 and all is well.
